Windows SDK: webgpu imported target missing IMPORTED_IMPLIB for non-Release configs

## Symptom
When a scaffolded Pulp plugin is configured via CMake on Windows using the published SDK + Visual Studio generator:

\`\`\`
cmake -S . -B build -G "Visual Studio 17 2022" -A x64 -DCMAKE_PREFIX_PATH=C:/Users/.../.pulp/sdk/0.2.0
\`\`\`

CMake fails with:

\`\`\`
CMake Error in CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "webgpu" configuration "MinSizeRel".

CMake Error in CMakeLists.txt:
  IMPORTED_IMPLIB not set for imported target "webgpu" configuration "RelWithDebInfo".
\`\`\`

## Root cause
The \`webgpu\` imported target exported by \`PulpConfig.cmake\` in the Windows SDK only populates \`IMPORTED_IMPLIB\` for the \`Release\` configuration. The Visual Studio generator requires all four configurations (Debug, Release, MinSizeRel, RelWithDebInfo) to have a valid implib, or the target must be marked \`IMPORTED_CONFIGURATIONS Release\` with the generator configured for Release only.

## Reproduction
1. Download v0.2.0 \`pulp-sdk-windows-x64.tar.gz\` (or use a local build)
2. Extract to \`~/.pulp/sdk/0.2.0/\`
3. Scaffold: \`pulp create MyPlugin --type bare --no-build\`
4. \`cd MyPlugin && cmake -S . -B build -G "Visual Studio 17 2022" -A x64 -DCMAKE_PREFIX_PATH=~/.pulp/sdk/0.2.0\`
5. See the error above

## Fix options
- **A.** Explicitly copy \`IMPORTED_IMPLIB_RELEASE\` to the other three configurations in the generated \`PulpConfig.cmake\` on Windows
- **B.** Set \`IMPORTED_CONFIGURATIONS Release\` on the \`webgpu\` target so CMake maps other configs to Release
- **C.** Use \`MAP_IMPORTED_CONFIG_<CONFIG>=Release\` on the target

Option B is the cleanest.

## Impact
Windows users cannot build a scaffolded plugin against the installed v0.2.0 SDK using the default Visual Studio generator. Workaround: use Ninja (\`-G Ninja\`) which is single-config.

## Discovered during
v0.2.0 Windows CLI verification (which is why it's being filed rather than blocking the release — the CLI itself works, only the downstream configure fails).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -780,10 +780,45 @@ if(PULP_HAS_WEBGPU AND DEFINED WEBGPU_RUNTIME_LIB AND TARGET webgpu)
         )
     endif()
 
-    if(WIN32 AND DEFINED WGPU_IMPLIB AND EXISTS "${WGPU_IMPLIB}")
-        install(FILES "${WGPU_IMPLIB}"
-            DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        )
+    # Install the wgpu_native import library alongside the runtime DLL.
+    #
+    # The WebGPU-distribution CMake sets WGPU_IMPLIB as a local variable
+    # inside FetchWgpuNativePrecompiled.cmake, so by the time execution
+    # returns to this file the variable is out of scope — the original
+    # guard `if(DEFINED WGPU_IMPLIB ...)` always evaluated to FALSE and
+    # the .lib was silently skipped from the SDK. That's why scaffolded
+    # Windows plugins fail with "IMPORTED_IMPLIB not set for imported
+    # target 'webgpu'" (see #94).
+    #
+    # Read the path from the imported target directly. Prefer the
+    # config-specific property (some WebGPU-distribution versions only
+    # populate IMPORTED_IMPLIB_RELEASE) and fall back to the plain
+    # IMPORTED_IMPLIB.
+    if(WIN32)
+        set(_pulp_wgpu_implib_src "")
+        get_target_property(_pulp_wgpu_implib_release webgpu IMPORTED_IMPLIB_RELEASE)
+        get_target_property(_pulp_wgpu_implib_plain webgpu IMPORTED_IMPLIB)
+        if(_pulp_wgpu_implib_release AND EXISTS "${_pulp_wgpu_implib_release}")
+            set(_pulp_wgpu_implib_src "${_pulp_wgpu_implib_release}")
+        elseif(_pulp_wgpu_implib_plain AND EXISTS "${_pulp_wgpu_implib_plain}")
+            set(_pulp_wgpu_implib_src "${_pulp_wgpu_implib_plain}")
+        elseif(DEFINED WGPU_IMPLIB AND EXISTS "${WGPU_IMPLIB}")
+            # Legacy fallback for older WebGPU-distribution versions.
+            set(_pulp_wgpu_implib_src "${WGPU_IMPLIB}")
+        endif()
+        if(_pulp_wgpu_implib_src)
+            install(FILES "${_pulp_wgpu_implib_src}"
+                DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            )
+        else()
+            message(WARNING
+                "Pulp: could not locate wgpu_native.lib for the Windows SDK "
+                "install. Downstream plugins will fail to link against webgpu. "
+                "Check that the webgpu imported target has IMPORTED_IMPLIB set.")
+        endif()
+        unset(_pulp_wgpu_implib_src)
+        unset(_pulp_wgpu_implib_release)
+        unset(_pulp_wgpu_implib_plain)
     endif()
 endif()
 

--- a/tools/cmake/PulpConfig.cmake.in
+++ b/tools/cmake/PulpConfig.cmake.in
@@ -43,6 +43,23 @@ if(PULP_LINUX)
     find_dependency(ALSA)
 endif()
 
+# The Pulp SDK is built and installed in the Release configuration only.
+# When a downstream plugin is built in Debug / MinSizeRel / RelWithDebInfo
+# (or the Visual Studio multi-config generator picks those configs), CMake
+# needs to know to use the Release configuration of Pulp's imported
+# targets. Without this, the Visual Studio generator errors with
+# "IMPORTED_IMPLIB not set for imported target 'webgpu' configuration
+# 'MinSizeRel'" etc.
+#
+# Set CMAKE_MAP_IMPORTED_CONFIG_<CONFIG>=Release for every non-Release
+# config unless the consuming project has already overridden it.
+foreach(_pulp_cfg Debug MinSizeRel RelWithDebInfo)
+    if(NOT DEFINED CMAKE_MAP_IMPORTED_CONFIG_${_pulp_cfg})
+        set(CMAKE_MAP_IMPORTED_CONFIG_${_pulp_cfg} "Release;")
+    endif()
+endforeach()
+unset(_pulp_cfg)
+
 set(_pulp_webgpu_runtime
     "${PULP_SDK_LIBRARY_DIR}/${PULP_SDK_SHARED_LIBRARY_PREFIX}wgpu_native${PULP_SDK_SHARED_LIBRARY_SUFFIX}")
 set(_pulp_webgpu_include_dir "${PULP_SDK_DIR}/external/webgpu/include")
@@ -50,20 +67,36 @@ set(_pulp_webgpu_include_dir "${PULP_SDK_DIR}/external/webgpu/include")
 if(NOT TARGET webgpu
    AND EXISTS "${_pulp_webgpu_runtime}"
    AND EXISTS "${_pulp_webgpu_include_dir}")
+
+    # On Windows, MSVC needs an import library (wgpu_native.lib) in
+    # addition to the runtime DLL. Fail fast with a clear message if the
+    # SDK install is broken (#94 was caused by this silently missing).
+    if(WIN32)
+        set(_pulp_webgpu_implib
+            "${PULP_SDK_LIBRARY_DIR}/${PULP_SDK_SHARED_LIBRARY_PREFIX}wgpu_native.lib")
+        if(NOT EXISTS "${_pulp_webgpu_implib}")
+            message(FATAL_ERROR
+                "Pulp SDK is missing wgpu_native.lib at ${_pulp_webgpu_implib}. "
+                "This is a broken SDK install — the runtime DLL is present "
+                "but the import library needed for linking is not. "
+                "Reinstall the Pulp SDK (v0.2.1 or newer) to pick up the fix "
+                "from https://github.com/danielraffel/pulp/issues/94.")
+        endif()
+    endif()
+
     add_library(webgpu SHARED IMPORTED)
     set_target_properties(webgpu PROPERTIES
         IMPORTED_LOCATION "${_pulp_webgpu_runtime}"
+        IMPORTED_LOCATION_RELEASE "${_pulp_webgpu_runtime}"
         INTERFACE_COMPILE_DEFINITIONS "WEBGPU_BACKEND_WGPU"
         INTERFACE_INCLUDE_DIRECTORIES
             "${_pulp_webgpu_include_dir};${_pulp_webgpu_include_dir}/webgpu"
     )
 
     if(WIN32)
-        set(_pulp_webgpu_implib
-            "${PULP_SDK_LIBRARY_DIR}/${PULP_SDK_SHARED_LIBRARY_PREFIX}wgpu_native.lib")
-        if(EXISTS "${_pulp_webgpu_implib}")
-            set_property(TARGET webgpu PROPERTY IMPORTED_IMPLIB "${_pulp_webgpu_implib}")
-        endif()
+        set_property(TARGET webgpu PROPERTY IMPORTED_IMPLIB "${_pulp_webgpu_implib}")
+        set_property(TARGET webgpu PROPERTY
+            IMPORTED_IMPLIB_RELEASE "${_pulp_webgpu_implib}")
     elseif(UNIX AND NOT APPLE)
         set_property(TARGET webgpu PROPERTY IMPORTED_NO_SONAME TRUE)
     endif()

